### PR TITLE
#575: qois: fix rank qois

### LINKED ISSUE #575: Post PR563 merge rendering error
## Steps to reproduce:
* use branch 130 of vt-tv
* run `synthetic-blocks.yaml` test case of LBAF's `develop` branch (post merger of PR #563 )

## Description:
The following vt-tv runtime error occurs:
```
---- Normalizing Edges for phase 1 ----
Size of all objects: 9
Traceback (most recent call last):
  File "/Users/pppebay/Documents/Git/LB-analysis-framework/src/lbaf/Applications/LBAF_app.py", line 674, in <module>
    LBAFApplication().run()
  File "/Users/pppebay/Documents/Git/LB-analysis-framework/src/lbaf/Applications/LBAF_app.py", line 648, in run
    vttv.tvFromJson(ranks_json_str, str(vttv_params), num_ranks)
IndexError: unordered_map::at: key not found
```

### PR DESCRIPTION
**Pull request by lifflander**
Fixes #575

### CODE REVIEW

**Review by pierrepebay — APPROVED** [review]

File: src/lbaf/IO/lbsVTDataWriter.py [hunk 1]
```diff
@@ -325,9 +325,22 @@ def _json_serializer(self, rank_phases_double) -> str:
...
                             continue
 
+                        rank_qois = it_r.get_qois()
+
                         # Add task data for current rank in iteration
                         iteration_data["tasks"] = self.__create_task_data(it_r)
 
+                        iteration_data["user_defined"] = {
+                            qoi_name: qoi_getter() for qoi_name, qoi_getter in rank_qois.items()
+                            if qoi_name != "homed_blocks_ratio" # omit for now because it might be nan
```
Comment (ppebay) [inline src/lbaf/IO/lbsVTDataWriter.py:335]: Put comment above

File: src/lbaf/Model/lbsRank.py [hunk 2]
```diff
@@ -272,7 +272,9 @@ def remove_migratable_object(self, o: Object):
     @qoi
     def get_load(self) -> float:
         """Return total load on rank."""
-        return sum(o.get_load() for o in self.__migratable_objects.union(self.__sentinel_objects))
+        val : float = 0.0
```
Comment (ppebay) [inline src/lbaf/Model/lbsRank.py:275]: General remark: if we use typing here we should be consistent and do it throughout the code

**Review by ppebay — CHANGES_REQUESTED** [review]

File: src/lbaf/IO/lbsVTDataWriter.py (hunk 1 above)
Comment (ppebay) [reply src/lbaf/IO/lbsVTDataWriter.py:336]: Done -- resolved

File: src/lbaf/Model/lbsRank.py (hunk 2 above)
Comment (ppebay) [reply src/lbaf/Model/lbsRank.py:275]: Follow-on issue #577 created

**Review by ppebay — APPROVED** [review]

**Pull request merged**